pkg/cmd/pipelinerun/describe_test.go

```diff
@@ -1212,3 +1212,122 @@ func TestPipelineRunDescribeV1beta1_taskrun_with_no_status(t *testing.T) {
 	}
 	golden.Assert(t, got, fmt.Sprintf("%s.golden", t.Name()))
 }
+
+func TestPipelineRunDescribe_last(t *testing.T) {
+	clock := clockwork.NewFakeClock()
+
+	trs := []*v1alpha1.TaskRun{
+		tb.TaskRun("tr-1", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(2*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(5*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+		tb.TaskRun("tr-2", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(5*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(9*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+		tb.TaskRun("tr-3", "ns",
+			tb.TaskRunStatus(
+				tb.TaskRunStartTime(clock.Now().Add(5*time.Minute)),
+				cb.TaskRunCompletionTime(clock.Now().Add(9*time.Minute)),
+				tb.StatusCondition(apis.Condition{
+					Type:   apis.ConditionSucceeded,
+					Status: corev1.ConditionTrue,
+				}),
+			),
+		),
+	}
+
+	/*
+		pipelinerun.GetAllPipelineRuns( ) returns a slice of strings containing all pipelineruns starting from the most recent one,
+		hence here the recent one is kept as the first element of the slice `pipelineRuns`
+	*/
+	pipelineRuns := []*v1alpha1.PipelineRun{
```

Comment:
I would search for how tests for API version `v1beta1` are done in this file and model your tests off the way PipelineRuns are included as test data in those. `v1alpha1` tests rely on test builders (i.e. `tb`), which are being moved away from.

Reply:
yaa, that would be helpful .. I too will search and let you know if I find something.

Reply:
Example: https://github.com/tektoncd/cli/blob/7f56bc9c76faf67dde19bb878ccc12166c0c67f5/pkg/cmd/pipelinerun/describe_test.go#L917